onnxruntime/core/providers/openvino/ov_versions/capabilities.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "core/framework/compute_capability.h"
+
+namespace onnxruntime {
+namespace openvino_ep {
+
+#if (defined OPENVINO_2020_2) || (defined OPENVINO_2020_3)
+std::vector<std::unique_ptr<ComputeCapability>>
+GetCapability_2020_2(const onnxruntime::GraphViewer& graph_viewer, const std::string device_id);
+
+#elif defined OPENVINO_2020_4
+std::vector<std::unique_ptr<ComputeCapability>>
+GetCapability_2020_4(const onnxruntime::GraphViewer& graph_viewer, const std::string device_id);
```

Comment:
Are we going to always copy the same code and make some modification for new version? I think there are quite a lot of share code. What are you going to do if bugs found for the shared code? Should we think about a way for code sharing. I suppose there would be less limitation/rules for newer version. Is it possible to create base class with basic check/validations for newer OpenVino version, the older one inherit from that and extend the check/validations.

Reply:
Yes, using inheritance is certainly a cleaner solution. Having fewer checks/restrictions with newer OpenVINO versions is generally true with only GetCapability code. Refactoring this is something that we are planning to do for the next release. 

For other areas of code (Compile and Infer functions) it gets a bit more complex as code working with newer OpenVINO versions does some things differently due to newer API features available. With this area, we are still considering on how to refactor the logic more cleanly.